src/main/java/com/google/javascript/gents/TypeAnnotationPass.java

```diff
@@ -147,37 +147,42 @@ public void visit(NodeTraversal t, Node n, Node parent) {
             break;
           }
           if (NodeUtil.isNameDeclaration(parent)) { // Variable declaration
-            if (bestJSDocInfo != null) {
-              setTypeExpression(n, bestJSDocInfo.getType(), false);
-            }
+            maybeSetTypeExpression(n, bestJSDocInfo, false);
           } else if (parent.isParamList()) { // Function parameters
             JSDocInfo parentDocInfo = NodeUtil.getBestJSDocInfo(parent.getParent());
-            if (parentDocInfo == null) {
-              break;
-            }
-            JSTypeExpression parameterType = parentDocInfo.getParameterType(n.getString());
-            if (parameterType != null) {
-              Node attachTypeExpr = n;
-              // Modify the primary AST to represent a function parameter as a
-              // REST node, if the type indicates it is a rest parameter.
-              if (parameterType.getRoot().getToken() == Token.ELLIPSIS) {
-                attachTypeExpr = IR.rest(n.getString());
-                nodeComments.replaceWithComment(n, attachTypeExpr);
-                compiler.reportCodeChange();
-              }
-              // Modify the AST to represent an optional parameter
-              if (parameterType.getRoot().getToken() == Token.EQUALS) {
-                attachTypeExpr = IR.name(n.getString());
-                attachTypeExpr.putBooleanProp(Node.OPT_ES6_TYPED, true);
-                nodeComments.replaceWithComment(n, attachTypeExpr);
-                compiler.reportCodeChange();
+            if (parentDocInfo != null) {
```

Comment:
It actually made a difference here (see old line 177). I refactored the code to make this more explicit.